modules/hal_nordic/CMakeLists.txt

```diff
@@ -1,8 +1,121 @@
+      ${CMAKE_CURRENT_SOURCE_DIR}/nrf_802154/serialization/platform/nrf_802154_spinel_log.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/nrf_802154/serialization/platform/nrf_802154_spinel_backend_ipc.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c
+  )
+endif ()
+
+if (CONFIG_NRF_802154_SER_RADIO)
+    target_sources(nrf-802154-platform
+      PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/nrf_802154/serialization/platform/nrf_802154_init_net.c
+    )
+endif ()
+
+target_compile_definitions(nrf-802154-platform
+  PUBLIC
+    # Radio driver shim layer uses raw api
+    NRF_802154_USE_RAW_API=1
+
+    # Number of slots containing short addresses of nodes for which
+    # pending data is stored.
+    NRF_802154_PENDING_SHORT_ADDRESSES=${CONFIG_NRF_802154_PENDING_SHORT_ADDRESSES}
+
+    # Number of slots containing extended addresses of nodes for which
+    # pending data is stored.
+    NRF_802154_PENDING_EXTENDED_ADDRESSES=${CONFIG_NRF_802154_PENDING_EXTENDED_ADDRESSES}
+
+    # Number of buffers in receive queue.
+    NRF_802154_RX_BUFFERS=${CONFIG_NRF_802154_RX_BUFFERS}
+
+    # CCA mode options
+    NRF_802154_CCA_CORR_LIMIT_DEFAULT=${CONFIG_NRF_802154_CCA_CORR_LIMIT}
+    NRF_802154_CCA_CORR_THRESHOLD_DEFAULT=${CONFIG_NRF_802154_CCA_CORR_THRESHOLD}
+    NRF_802154_CCA_ED_THRESHOLD_DEFAULT=${CONFIG_NRF_802154_CCA_ED_THRESHOLD}
+
+    # Enable CSMA/CA
+    NRF_802154_CSMA_CA_ENABLED=1
+    NRF_802154_TX_STARTED_NOTIFY_ENABLED=1
+
+    # ACK timeout
+    NRF_802154_ACK_TIMEOUT_ENABLED=1
+)
+
+if (CONFIG_NRF_802154_CCA_MODE_ED)
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_CCA_MODE_DEFAULT=NRF_RADIO_CCA_MODE_ED)
+elseif (CONFIG_NRF_802154_CCA_MODE_CARRIER)
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_CCA_MODE_DEFAULT=NRF_RADIO_CCA_MODE_CARRIER)
+elseif (CONFIG_NRF_802154_CCA_MODE_CARRIER_AND_ED)
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_CCA_MODE_DEFAULT=NRF_RADIO_CCA_MODE_CARRIER_AND_ED)
+elseif (CONFIG_NRF_802154_CCA_MODE_CARRIER_OR_ED)
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_CCA_MODE_DEFAULT=NRF_RADIO_CCA_MODE_CARRIER_OR_ED)
+endif()
+
+if (NOT CONFIG_IEEE802154_NRF5 AND NOT CONFIG_IEEE802154_NRF5_EXT_IRQ_MGMT)
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=1)
+else()
+  target_compile_definitions(nrf-802154-platform PUBLIC NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=0)
+endif()
+
+set(NRF52_SERIES  ${CONFIG_SOC_SERIES_NRF52X}        CACHE INTERNAL "")
+set(NRF53_SERIES  ${CONFIG_SOC_SERIES_NRF53X}        CACHE INTERNAL "")
+set(SERIALIZATION ${CONFIG_NRF_802154_SERIALIZATION} CACHE INTERNAL "")
+set(SER_HOST      ${CONFIG_NRF_802154_SER_HOST}      CACHE INTERNAL "")
+set(RADIO_DRIVER  ${CONFIG_NRF_802154_RADIO_DRIVER}  CACHE INTERNAL "")
+set(SL_OPENSOURCE ${CONFIG_NRF_802154_SL_OPENSOURCE} CACHE INTERNAL "")
```

Comment:
You still have a set of cache variables here.

Reply:
Now those variables are no longer cached.